### ISSUE TITLE
When the input method is Chinese, it can be input in the non-editing state

### ISSUE THREAD

**Issue by AKclown** [replit/codemirror-vim#66]
Steps to reproduce：
On the replit editor website, open the vim mode, exit the vim editor state, and you can still enter Chinese characters
https://replit.com/@AKclown/AKclown#index.js

Desired result：
In the non-editing state, it should not be possible to input Chinese characters

![image](https://user-images.githubusercontent.com/33648653/203461895-a992e7d6-1f47-4f01-b893-8b87628db963.png)

**Comment by yiwkd2** [comment]
I found the same issue for Korean, and I think I figured out what the cause of this issue is.
Please refer to this article.
[https://forum.obsidian.md/t/vim-keybinding-does-not-support-non-alphabetic-languages/49566](url)

**Comment by sergeichestakov** [comment]
wow, thanks for the report! that's quite the bug. @nightwing mind taking a look? is it possible our logic preventing insertion when not in insert mode is not accounting for non-alphabetic languages like Chinese and Korean?

in the meantime, we accept PRs if you find the fix before we can get to it 🙂

**Comment by yiwkd2** [comment]
I observed the following while trying to fix this by myself.

- In normal mode, the input of non-alphabetic languages generates CompositionEvent such as compositionstart, compositionupdate, etc rather than keypress or keydown.
- I've tried to cancel compositionstart event by creating a new event handler but didn't work.

As a workaround, I found [im-select](https://github.com/daipeihust/im-select), and this tool is being used by [VSCodeVim](https://github.com/VSCodeVim/Vim) and [vim-im-select-obsidian](https://github.com/ALONELUR/vim-im-select-obsidian). And I made [conversion-select](https://github.com/yiwkd2/conversion-select) for me and Koreans who are using Windows.

These tools change the keyboard itself or conversion status of IME on vim mode changes to prevent typing non-alphabetic languages in normal mode from happening.
I'm not sure if this mechanism can be merged into this code mirror or not.

**Comment by nightwing** [comment]
The linked pull request disables input in normal mode, and treats keys as their english equivalent when browser provides the information for that. Please check if https://raw.githack.com/replit/codemirror-vim/github-demo/dev/web-demo.html works as expected.
